src/backend/distributed/commands/multi_copy.c

```diff
@@ -1751,6 +1752,12 @@ CitusCopyDestReceiverStartup(DestReceiver *dest, int operation,
 	LockShardListMetadata(shardIntervalList, ShareLock);
 	LockShardListResources(shardIntervalList, ShareLock);
 
+	/*
+	 * We might have some concurrent metadata changes and we need to accept the cache
+	 * invalidation messages to learn about the changes.
+	 */
+	AcceptInvalidationMessages();
```

Comment:
We should reload the cache entry below, i.e. `cacheEntry = DistributedTableCacheEntry(...);`.